src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,305 @@
+# Defines InterfaceValues and common methods
+"""
+    InterfaceValues(grid::AbstractGrid, quad_rule::FaceQuadratureRule, func_interpol_a::Interpolation, [geom_interpol_a::Interpolation], [func_interpol_b::Interpolation], [geom_interpol_b::Interpolation])
+
+An `InterfaceValues` object facilitates the process of evaluating values, averages, jumps and gradients of shape functions
+and function on the interfaces of finite elements.
+
+**Arguments:**
+
+* `grid` : instance of the current grid.
+* `quad_rule_a`: an instance of a [`FaceQuadratureRule`](@ref) for element A.
+* `quad_rule_b`: an instance of a [`FaceQuadratureRule`](@ref) for element B.
+* `func_interpol_a`: an instance of an [`Interpolation`](@ref) used to interpolate the approximated function for element A.
+* `func_interpol_b`: an instance of an [`Interpolation`](@ref) used to interpolate the approximated function for element B.
+* `geom_interpol_a`: an optional instance of an [`Interpolation`](@ref) which is used to interpolate the geometry for element A.
+  It uses the default interpolation of the respective [`RefShape`](@ref) by default.
+* `geom_interpol_b`: an optional instance of an [`Interpolation`](@ref) which is used to interpolate the geometry for element B.
+  It uses the default interpolation of the respective [`RefShape`](@ref) by default.
+ 
+**associated methods:**
+
+* [`shape_value_average`](@ref)
+* [`shape_value_jump`](@ref)
+* [`shape_gradient_average`](@ref)
+* [`shape_gradient_jump`](@ref)
+
+**Common methods:**
+
+* [`reinit!`](@ref)
+* [`getnquadpoints`](@ref)
+* [`getdetJdV`](@ref)
+
+* [`shape_value`](@ref)
+* [`shape_gradient`](@ref)
+* [`shape_divergence`](@ref)
+* [`shape_curl`](@ref)
+
+* [`function_value`](@ref)
+* [`function_gradient`](@ref)
+* [`function_symmetric_gradient`](@ref)
+* [`function_divergence`](@ref)
+* [`function_curl`](@ref)
+* [`spatial_coordinate`](@ref)
+"""
+InterfaceValues
+
+struct InterfaceValues{FVA, FVB, dim, C} <: AbstractValues
+    face_values_a::FVA
+    face_values_b::FVB
+    # used for quadrature point syncing
+    grid::Grid{dim, C, Float64}
+    cell_a_idx::ScalarWrapper{Int}
+    cell_b_idx::ScalarWrapper{Int}
```

Comment:
Can you elaborate on this design why we need this specific grid type and the full grid?

Reply:
Type stability, maybe I can change it like FVA/FVB

Reply:
Actually this might harm type stability in case of mixed grids (and if including the grid, not having the flexibility in the grid type might harm extensibility)